src/backend/looped/vehicle/infinite_veh_ammo.cpp

```diff
@@ -0,0 +1,40 @@
+#include "backend/looped_command.hpp"
+#include "natives.hpp"
+
+namespace big
+{
+	class infinite_veh_ammo : looped_command
+	{
+		using looped_command::looped_command;
+
+		virtual void on_tick() override
+        	{
+			if (self::veh && VEHICLE::DOES_VEHICLE_HAVE_WEAPONS(self::veh))
+            		{
+				for (int i = 0; i < 3; i++)
+				{
+					VEHICLE::SET_VEHICLE_WEAPON_RESTRICTED_AMMO(self::veh, i, -1);
+				}
+
+				VEHICLE::SET_VEHICLE_BOMB_AMMO(self::veh, -1);
+				VEHICLE::SET_VEHICLE_COUNTERMEASURE_AMMO(self::veh, -1);
+            		}
+        	}
+
+		virtual void on_disable() override
+		{
+			if (self::veh && VEHICLE::DOES_VEHICLE_HAVE_WEAPONS(self::veh))
+			{
+				for (int i = 0; i < 3; i++)
+				{
+					VEHICLE::SET_VEHICLE_WEAPON_RESTRICTED_AMMO(self::veh, i, 20);
```

Comment:
Are they always capped at `20`?

Reply:
I chose 20 because that's the missile capacity of the Op Mk2, but it's not necessarily the capacity of other vehicle weapon ammo. For example, the Deluxo can hold up to 30. I wasn't able to find a convenient method of pulling the vehicle weapon's ammo capacity from the index, which is what I'd prefer to do for a "Refill Vehicle Ammo" feature.

We could potentially set ammo capacity based on the vehicle itself, but that would require maintaining a database of all vehicles with weapons.

I think that just giving a flat quantity of 20 to all vehicle weapons after disabling Vehicle Infinite Ammo is a good compromise.

Reply:
Alright that's fine by me.